### ISSUE TITLE
请教一下通过parc改进mobilenet部分

### ISSUE THREAD

**Issue by FeiXie8** [hkzhang91/EdgeFormer#23]
您好，实在不好意思打扰一下，有两个问题希望请教一下
（1）请问论文中通过parc改进mobilenet的操作是用代码中的gcc_ca_mf_block去把mobilenet中的InvertedResidualBlock全部给替换了吗，或是其他做法呢？
（2）gcc_ca_mf_block、gcc_dk_mf_block、gcc_dk_ca_mf_block，这三个parc-block，哪一个是论文结果中用来改进Mobilenet等网络的block呢？
十分感谢！

**Comment by hkzhang-git** [comment]
改进Mobilenetv2 的做法是把mobilenetv2 中conv 替换为parc conv. 类似改动convnext 那样。gcc_ca_mf_block、gcc_dk_mf_block、gcc_dk_ca_mf_block 这些是一些后续扩展工作的半成品，动态parc 等一些操作。

**Comment by FeiXie8** [comment]
> 改进Mobilenetv2 的做法是把mobilenetv2 中conv 替换为parc conv. 类似改动convnext 那样。gcc_ca_mf_block、gcc_dk_mf_block、gcc_dk_ca_mf_block 这些是一些后续扩展工作的半成品，动态parc 等一些操作。

我不知道我在源代码找到的parc conv对不对希望能和您确认一下，
x_1_1 = F.conv2d(torch.cat((x_1, x_1[:, :, :-1, :]), dim=2), weight=K_1_H, bias=self.meta_1_H_bias, padding=0, groups=self.dim)是parc-H,
而x_2_1 = F.conv2d(torch.cat((x_2, x_2[:, :, :, :-1]), dim=3), weight=K_1_W, bias=self.meta_1_W_bias, padding=0, groups=self.dim)是Parc-W
请问这样理解是对的吗

**Comment by FeiXie8** [comment]
> 改进Mobilenetv2 的做法是把mobilenetv2 中conv 替换为parc conv. 类似改动convnext 那样。gcc_ca_mf_block、gcc_dk_mf_block、gcc_dk_ca_mf_block 这些是一些后续扩展工作的半成品，动态parc 等一些操作。

还有个问题想请教一下，改动convnext在图里面是替换成了两个gcc-w，但是按照论文的思路不是应该是一个gcc-h,和一个gcc-v吗

**Comment by FeiXie8** [comment]
我可能也希望通过parc改进一下自己的网络，所以十分好奇mobilenet里的改进方式，这一部分的源码请问可以在后面更新中看到吗

**Comment by hkzhang-git** [comment]
> > 改进Mobilenetv2 的做法是把mobilenetv2 中conv 替换为parc conv. 类似改动convnext 那样。gcc_ca_mf_block、gcc_dk_mf_block、gcc_dk_ca_mf_block 这些是一些后续扩展工作的半成品，动态parc 等一些操作。
> 
> 还有个问题想请教一下，改动convnext在图里面是替换成了两个gcc-w，但是按照论文的思路不是应该是一个gcc-h,和一个gcc-v吗

这个是画错了，是一个gcc-h, 和一个gcc-v. 对应各种改版的代码，很快会更新的。具体结构可以参照这个图。
![image](https://user-images.githubusercontent.com/41274618/190045916-d3b5daa7-4604-43fe-b8d6-f961fc1359a4.png)

**Comment by FeiXie8** [comment]
感谢解答！
